### PR TITLE
Check live datasets checksums too

### PR DESCRIPTION
**Pull request by ccl-core**
Raise ValueError also for live datasets when hashes of downloaded files do not match

### CODE REVIEW

**Comment by ccl-core** [comment]
Addresses #494

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/download_test.py [hunk 1]
```diff
@@ -160,28 +160,49 @@ def test_sha256_hashes_do_match(conforms_to, hash_value):
...
+def test_hashes_are_checked_for_live_datasets(dummy_ctx):
+    with tempfile.NamedTemporaryFile(delete=False) as f:
+        filepath = f.name
+        metadata = Metadata(ctx=dummy_ctx, name="bar")
+        file_object = create_test_file_object(
+            ctx=dummy_ctx,
+            name="foo",
+            content_url=os.fspath(filepath),
+            # Hash won't match.
+            sha256="12345",
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/download_test.py:200]: Parametrize the test for sha256 and md5?

**Review by marcenacp — APPROVED** [review]
Thanks!

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/download.py [hunk 2]
```diff
@@ -148,6 +148,13 @@ class Download(Operation):
     def _check_hash(self, filepath: epath.Path):
         if filepath.is_dir():
             return
+        ctx = self.node.ctx
+        # For live datasets which do not specify checksums, we do not check the hash.
+        if ctx.is_live_dataset and not (self.node.md5 or self.node.sha256):
+            logging.warning(
```
[the remark this answers is not in the document]
Comment (ccl-core) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/download.py:154]: I don't know how to test this properly though, I think pytest only captures logging from warning to above?

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/download_test.py (hunk 1 above)
Comment (ccl-core) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/download_test.py:200]: Done.

**Pull request merged**